arviz/tests/base_tests/test_data.py

```diff
@@ -1077,6 +1077,20 @@ def test_dict_to_dataset():
     assert set(dataset.b.coords) == {"chain", "draw", "c"}
 
 
+def test_nested_dict_to_dataset():
+    datadict = {
+        "top": {"a": np.random.randn(100), "b": np.random.randn(1, 100, 10)},
+        "d": np.random.randn(100),
+    }
+    dataset = convert_to_dataset(datadict, coords={"c": np.arange(10)}, dims={("top", "b"): ["c"]})
```

Comment:
I would extend the docstring, both parameter description and examples to show this. Mainly, how will flattened keys look like, and how can one specify coordinates/dimensions for nested variables. Otherwise we'd require users to read through the source code and/or tests in order to properly be able to use the function.